pkg/util/env.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+func MergeEnvLists(envLists ...[]corev1.EnvVar) []corev1.EnvVar {
+	resultList := make([]corev1.EnvVar, 0)
+	for _, list := range envLists {
+		for _, env := range list {
+			idx := FindEnvIndex(resultList, env.Name)
```

Comment:
not critical but we can use built-in `slices.IndexFunc` here

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/986/commits/75cdf5a120a01f03fc3d185a607d3cc1444fdb3f